controllers/operator/configmap.go

```diff
@@ -512,8 +513,12 @@ type registrationJSONData struct {
 func (r *AuthenticationReconciler) authIdpConfigMap(instance *operatorv1alpha1.Authentication, scheme *runtime.Scheme) *corev1.ConfigMap {
 	reqLogger := log.WithValues("Instance.Namespace", instance.Namespace, "Instance.Name", instance.Name)
 	isPublicCloud := isPublicCloud(r.Client, instance.Namespace, "ibmcloud-cluster-info")
-	clusterAddress := getClusterAddress(r.Client, instance.Namespace, "ibmcloud-cluster-info")
-	if strings.EqualFold(clusterAddress, "") {
+	clusterAddress, cerr := getClusterAddress(r.Client, instance.Namespace, "ibmcloud-cluster-info")
+	if cerr != nil {
+		reqLogger.Error(cerr, "Failed to fetch the cluster_address from the ibmcloud-cluster-info ConfigMap")
+		return nil
```

Comment:
If this prompts a return of nil, that means that we need to check for nil in all of the cases where we are accessing fields on the return value of this ConfigMap. This is an oversight that's present throughout the handleConfigMap function in that update portion where the currentConfigMap is having its values set to the newConfigMap ones.